.github/workflows/_build-wheels-impl.yaml

```diff
@@ -31,6 +31,17 @@ jobs:
           os: ${{ inputs.os }}
       - name: Clean up dirty files
         run: git clean -f -d
+      - name: Set Python Version
+        id: python-version
+        run: |
+          if [[ "${{ inputs.os }}" == "ubuntu-20.04" ]]; then
```

Comment:
Generally not a fan of hiding implementation details in something like this, given that we take in `os` as an argument, but given what I said above, this is probably what I'd do, too